.pre-commit-config.yaml

```diff
@@ -1,35 +1,22 @@
+exclude: |
+    (?x)^(
+        versioneer\.py|
+        theano/_version\.py|
+        doc/.*|
+        bin/.*
+    )$
 repos:
   - repo: https://github.com/psf/black
     rev: 20.8b1
     hooks:
       - id: black
         language_version: python3
-        exclude: |
-            (?x)^(
-                versioneer\.py|
-                theano/_version\.py|
-                doc/.*|
-                bin/.*
-            )$
   - repo: https://gitlab.com/pycqa/flake8
     rev: 3.8.4
     hooks:
       - id: flake8
-        exclude: |
-            (?x)^(
-                versioneer\.py|
-                theano/_version\.py|
-                doc/.*|
-                bin/.*
-            )$
+        exclude: ^theano/gof/unify\.py$
```

Comment:
unify_walk is redefined loads of time

Reply:
`exclude` in `setup.cfg` only has an effect if you run `flake8` on a directory though (see https://github.com/pre-commit/pre-commit/issues/1723) so it needs to be set here, at least until it's removed as part of the refactoring

Reply:
After having looked at that issue, it seems like the complete answer is that `pre-commit` has chosen not to implement functionality that could make use of the `exclude` information from `setup.cfg` (or perhaps anywhere else).  The result is that _`pre-config`_ is too simple and will only apply `flake8` to each file individually unless you use its configuration parameters.

The real problem is that `pre-commit` is taking over the responsibility of executing `flake8` over directories/multiple-files from `flake8` itself, but it's not honoring the settings that `flake8` accepts when it performs this task&mdash;even when the settings are exactly the same between the two (e.g. `exclude` in `setup.cfg` has literally the same meaning and purpose as `exclude` in `.pre-commit-config.yaml`).

Regardless, the result is that one has to maintain both a new tool&mdash;with its own peculiarities&mdash;_and_ the same setting(s) in two different places, which is never good.

Reply:
It _is_ possible to get it to respect `setup.cfg` by using `pass_filenames: false`. Then it'll work the same as if you just ran `flake8` locally, and `setup.cfg` will be respected. I can do that if you want, I just usually prefer to have it run on each file changed individually so that when I make a commit, it'll only run on the files I've staged and so it'll be really fast and I'll get faster feedback.

> Regardless, the result is that one has to maintain both a new tool—with its own peculiarities—and the same setting(s) in two different places, which is never good.

Or alternatively, you could run linters/formatters via pre-commit exclusively and then only need to put settings in one place. But if we use `pass_filenames: false` as above, then that won't be necessary - shall I change it to that then?

Reply:
> Then it'll work the same as if you just ran `flake8` locally, and `setup.cfg` will be respected.

That seems hardly worth the change to `pre-commit` in the first place, no?

> it'll only run on the files I've staged and so it'll be really fast and I'll get faster feedback.

As long as the upkeep cost isn't greater than this benefit, I'm fine with it.